### ISSUE TITLE
The links to the applications should point directly to them, with no intermediate step

### ISSUE THREAD

**Issue by SergioArbarviro** [michaellaunay/alirpunkto#142]
### Current state
The list of applications to which AlirPunkto gives access to currently is given as the list of URLs of the main page of the application. 
As a consequence, if these applications have other means to access them as the SSO that AlirPunkto feeds, the user needs to select the SSO as the entry point to acutally enter the application, which is not trivial to understand when s/he legitimately expects the click on the list provided by AlirPunkto to lead to the application itself.

### Desired state
The click on the URL provided by AlirPunkto should lead directly to the "inside" of the application, after the login.

In order to achieve this, the links provided by AlirPunkto should be, for each application, the URLs of the login buttons using the SSO. In the specific case of the pilot case of implementing AlirPunkto, these URLs are:

- for NextCloud: [https://workspace.cosmopolitical.coop/apps/sociallogin/custom_oidc/Keycloak](https://workspace.cosmopolitical.coop/apps/sociallogin/custom_oidc/Keycloak);
- for Moodle: [https://learn.cosmopolitical.coop/auth/oauth2/login.php?id=1&wantsurl=%2F&sesskey=2Y5n4rlEvC](https://learn.cosmopolitical.coop/auth/oauth2/login.php?id=1&wantsurl=%2F&sesskey=2Y5n4rlEvC); 
- for Drupal: [https://www.cosmopolitical.coop/moLogin?appname=CosmoPolitical%20Cooperative%20login](https://www.cosmopolitical.coop/moLogin?appname=CosmoPolitical%20Cooperative%20login)

**Comment by micklamenace** [comment]
The Moodle url fails !

**Comment by ldidry** [comment]
The problem with the Moodle URL is the `sesskey` argument.

Unfortunately, it seems that this argument is required by Moodle, and is (obviously) modified each time you go to the login page.

**Comment by SergioArbarviro** [comment]
I hence propose for Moodle the following URL: [https://learn.cosmopolitical.coop/login/index.php](https://learn.cosmopolitical.coop/login/index.php), with an additional text informing that the user should click on "CosmoPolitical login".

**Comment by SergioArbarviro** [comment]
The link for Drupal has been changed in the text of the issue. Please take the new one into consideration.

**Comment by SergioArbarviro** [comment]
I just changed the link for **LiquidFeedback** in the text of the issue. Please take the new one into consideration.

**Comment by SergioArbarviro** [comment]
I have added the link and the text for **SYMPA** in the text of the issue. Please take the new one into consideration.

**Comment by michaellaunay** [comment]
The link to drupal fails

**Comment by michaellaunay** [comment]
Only the new Sympa link and operational decisions link are succeeding.

**Comment by michaellaunay** [comment]
Links that work :
https://workspace.cosmopolitical.coop/apps/sociallogin/custom_oidc/Keycloak
https://www.cosmopolitical.coop/mo_login (but with a message on the page)
https://lists.cosmopolitical.coop/sympa
https://operationaldecisions.cosmopolitical.coop/

Links that don't work :
https://learn.cosmopolitical.coop/login/index.php

Not yet implemented :
https://operationaldecisions.cosmopolitical.coop/

I propose switching login and sso_login to make sso_login the default login method.

**Comment by ldidry** [comment]
As a said in a mail, `https://workspace.cosmopolitical.coop/apps/sociallogin/custom_oidc/Keycloak` can’t be used because if you go to that URL while being already logged in Nextcloud, you got an error message.

**Comment by ldidry** [comment]
And I [found a setting for Social login](https://github.com/zorn-v/nextcloud-social-login/issues/296#issuecomment-912138561): `'social_login_auto_redirect' => true`.

So we could just use https://workspace.cosmopolitical.coop as NC’s URL.

And to use the login form, you just have to add `?noredir=1` to https://workspace.cosmopolitical.coop/login.

Should I activate the setting?

**Comment by michaellaunay** [comment]
I changed the workspace link to https://workspace.cosmopolitical.coop and that works.

**Comment by SergioArbarviro** [comment]
I have activated the setting in the Social Login app of NextCloud that "hides the default login" - hence having the default login page be that displayed on  https://github.com/michaellaunay/alirpunkto/issues/142#issuecomment-2203596395

@ldidry Please activate that setting that you mention in https://github.com/michaellaunay/alirpunkto/issues/142#issuecomment-2203623404

**Comment by ldidry** [comment]
@SergioArbarviro Setting activated.

**Comment by SergioArbarviro** [comment]
After having seen how it works, I prefer to keep it **without** this automatic redirection. Seeing the login page of the Worskspace website confirms to the user that s/he is at the right place.

@ldidry Please revert to the original setting = without the redirection.

**Comment by ldidry** [comment]
Setting deactivated.

**Comment by SergioArbarviro** [comment]
To summarise where we stand, the links should be:

- for NextCloud: [https://workspace.cosmopolitical.coop](https://workspace.cosmopolitical.coop) - (PRESENT)  **with a text informing the user that s/he needs to click on the "CosmoPolitical login" button (TO BE ADDED)**;
- for Moodle: [https://learn.cosmopolitical.coop/login/index.php](https://learn.cosmopolitical.coop/login/index.php) - with a text informing the user that s/he needs to click on the "CosmoPolitical login" button - PRESENT;
- for Drupal: [https://www.cosmopolitical.coop/user/login](https://www.cosmopolitical.coop/user/login) - **(TO BE MODIFIED)**
- for LiquidFeedback: [https://operationaldecisions.cosmopolitical.coop/index/index.html](https://operationaldecisions.cosmopolitical.coop/index/index.html) - (PRESENT);
- for SYMPA: [https://authentication.cosmopolitical.coop/realms/cosmopolitical-login/protocol/cas/login?service=https://lists.cosmopolitical.coop/sympa/sso_login_succeeded/CosmoPolitical_login](https://authentication.cosmopolitical.coop/realms/cosmopolitical-login/protocol/cas/login?service=https://lists.cosmopolitical.coop/sympa/sso_login_succeeded/CosmoPolitical_login) - (PRESENT);
- for KuneAgi: to be determined once the connection of KuneAgi to the SSO has been implemented.

**Comment by ldidry** [comment]
> **with a text informing the user that s/he needs to click on the "CosmoPolitical login" button (TO BE ADDED)**;

The button, for now, says "Log in with CosmoPolitical login". There is a setting to remove the "Log in with", and the "CosmoPolitical login" text is a setting we can change at will.

So, a simple solution would be to remove the "Log in with" and change the button text to something else.

Con: no internationalisation, it would be the same text for all languages.

Or we could use the [login notes app](https://apps.nextcloud.com/apps/login_notes). I just installed it and added a text to say to use the big green button to log in. Still: no internationalisation, though.

**Comment by SergioArbarviro** [comment]
In the **test infrastructure**, the list of sites after login still contains dummy 'example.com' sites, not the actual ones  used by the infrastructure. 

If this is the intended behaviour, then no problem, otherwise, it means that the commit doesn't fix the issue. I hence **re-open** it.

**Comment by michaellaunay** [comment]
I'm updating the `production.ini` file and restarting the container.

Le samedi 25 juillet 2026 à 14:34 -0700, SergioArbarviro a écrit :

**Comment by SergioArbarviro** [comment]
Indeed, now a list is present, which however does **not** match the list provided above in https://github.com/michaellaunay/alirpunkto/issues/142#issuecomment-2508892617. The links and texts needing **updating** are:

- the link to the **Workspace** +  the message whereby the user should connect by simply clicking on the "CosmoPolitical login" button;
- the text accompanying the link to the **Presentation page** should be modified to: "Our presentation portal, where you can make donations or purchase training materials and track your account. If you are Cooperator or have just registered as a Cooperator, you can in addition on this site: (a) purchase  shares of the Cooperative and (b) pay your yearly contribution."

The text is in English only. **Internationalisation** would be a next step, but as **enhancement**, not as part of this issue.
